web/packages/teleterm/src/ui/DocumentCluster/useUserPreferences.ts

```diff
@@ -77,12 +77,17 @@ export function useUserPreferences(clusterUri: ClusterUri): {
   const [initialFetchAttempt, runInitialFetchAttempt] = useAsync(
     useCallback(
       async () =>
-        retryWithRelogin(appContext, clusterUri, () =>
-          appContext.tshd.getUserPreferences(
+        retryWithRelogin(appContext, clusterUri, async () => {
+          const { response } = await appContext.tshd.getUserPreferences(
             { clusterUri },
-            cloneAbortSignal(initialFetchAttemptAbortController.current.signal)
-          )
-        ),
+            {
+              abort: cloneAbortSignal(
+                initialFetchAttemptAbortController.current.signal
+              ),
+            }
+          );
+          return response.userPreferences;
```

Comment:
Pretty unfortunate that the calls can no longer use this compact form. I suppose in the future we could think of adding a helper function like `extractResponse` which accepts an optional property name so that we can write code like this:

```typescript
  const [initialFetchAttempt, runInitialFetchAttempt] = useAsync(
    useCallback(
      () =>
        retryWithRelogin(appContext, clusterUri, () =>
          appContext.tshd
            .getUserPreferences(
              { clusterUri },
              {
                abort: cloneAbortSignal(
                  initialFetchAttemptAbortController.current.signal
                ),
              }
            )
            .then(extractResponse)
            // or .then(extractResponse('userPreferences'))
            // depending on what is more desirable in the given context
        ),
      [appContext, clusterUri]
    )
  );
```

OTOH now that I compare this more "compact" form that I suggested here, the current one isn't really that bad.

Reply:
Hmm yeah, I feel that `extractResponse` doesn't change much. 
The new form is minimally more verbose, but personally I'm okay with that.